test/unittest/xmlparser/XMLProfileParserTests.cpp

```diff
@@ -998,6 +1313,156 @@ TEST_F(XMLProfileParserTests, skip_default_xml)
     EXPECT_NE(participant_atts_none.domainId, participant_atts_default.domainId);
 }
 
+/*
+ * Tests whether the FASTRTPS_DEFAULT_PROFILES_FILE environment file correctly loads the selected file as default.
+ * - participant_atts_default contains the attributes in the default file in this folder.
+ * - participant_atts_file contains the attributes in the default file created by the test.
+ */
+TEST_F(XMLProfileParserTests, default_env_variable)
+{
+    const char* xml =
+            "                                                                                                                  \
+        <profiles>                                                                                                     \
+            <participant profile_name=\"test_participant_profile\" is_default_profile=\"true\">                        \
+                <domainId>20203011</domainId>                                                                          \
+                <rtps></rtps>                                                                                          \
+            </participant>                                                                                             \
+        </profiles>                                                                                                    \
+    ";
+    tinyxml2::XMLDocument xml_doc;
+    xml_doc.Parse(xml);
+    xml_doc.SaveFile("FASTRTPS_PROFILES.xml");
+
+    ParticipantAttributes participant_atts_default;
+    xmlparser::XMLProfileManager::loadDefaultXMLFile();
+    xmlparser::XMLProfileManager::getDefaultParticipantAttributes(participant_atts_default);
+
+#ifdef _WIN32
+    _putenv_s("FASTRTPS_DEFAULT_PROFILES_FILE", "FASTRTPS_PROFILES.xml");
+#else
+    setenv("FASTRTPS_DEFAULT_PROFILES_FILE", "FASTRTPS_PROFILES.xml", 1);
+#endif // ifdef _WIN32
+    ParticipantAttributes participant_atts_file;
+    xmlparser::XMLProfileManager::loadDefaultXMLFile();
+    xmlparser::XMLProfileManager::getDefaultParticipantAttributes(participant_atts_file);
+    remove("FASTRTPS_PROFILES.xml");
+
+    EXPECT_NE(participant_atts_file.domainId, participant_atts_default.domainId);
+}
+
+/*
+ * Tests the return codes of the loadXMLFile method.
+ *  1. On parsing a correctly parsed standalone profiles file, expect XMLP_ret::XML_OK.
+ *  2. On parsing a correctly parsed rooted profiles file, expect XMLP_ret::XML_OK.
+ *  3. On parsing an incorrect element inside profiles element, expect XMLP_ret::XML_ERROR.
+ *  4. On giving an empty string as a filename, expect XMLP_ret::XML_ERROR.
+ *  5. On parsing the same file twice, expect XMLP_ret::XML_OK.
+ *  6. On parsing an incorrect element inside profiles element, expect XMLP_ret::XML_ERROR.
+ */
+TEST_F(XMLProfileParserTests, loadXMLFile)
+{
+    const char* filename = "minimal_file.xml";
+    tinyxml2::XMLDocument xml_doc;
+
+    {
```

Comment:
Same as before. Brief comment for each block. You can copy and paste the comments above in its corresponding block